language/bytecode_verifier/src/type_memory_safety.rs

```diff
@@ -51,8 +51,24 @@ impl<'a> TypeAndMemorySafetyAnalysis<'a> {
         let locals_signature_view = function_definition_view.locals_signature();
         let function_signature_view = function_definition_view.signature();
         let mut locals = BTreeMap::new();
+        if function_signature_view.arg_count() > locals_signature_view.len() {
```

Comment:
Todd had mentioned that it may be best to perform this check in a different pass e.g. a the bounds checking pass. What do you think @tnowacki?

Reply:
In a private conversation, we decided to leave this check in type_memory_safety.rs.

Reply:
I think it's fine to have it here and personally I like it here more than the bounds checking. Or more precisely, I am not sure that is a bounds check strictly speaking. And type safety seems good to me. 
Also maybe this is good for now until we decide what to do with args and locs?
thoughts?

Reply:
I'm fine with it here, but consider another candidate: the signature checker.